# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/thunderbird/hotfix-no-pip-check.patch
# Copyright (C) 2024 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Disable pip checks as mach is broken for python 3.12 and up.
It checks versions of unrelated packages.

We already provide python packages at system level anyway so
the overhead of extra checks is not needed.

  - NoTag <notag@t2sde.org>

--- firefox-130.0.1/python/mach/mach/site.py.vanilla	2024-09-17 17:44:02.205297850 +0200
+++ firefox-130.0.1/python/mach/mach/site.py	2024-09-17 17:55:24.758627113 +0200
@@ -1368,18 +1368,6 @@
             f.write("\n".join(pthfile_lines))
 
         pip = pip_command(python_executable=check_env.python_path)
-        if requirements:
-            packages = _resolve_installed_packages(check_env.python_path)
-            validation_result = RequirementsValidationResult.from_packages(
-                packages, requirements
-            )
-            if not validation_result.has_all_packages:
-                subprocess.check_call(pip + ["list", "-v"], stdout=sys.stderr)
-                print(validation_result.report(), file=sys.stderr)
-                raise Exception(
-                    f'The "{virtualenv_name}" site is not compatible with the installed '
-                    "system Python packages."
-                )
 
         check_result = subprocess.run(
             pip + ["check"],
@@ -1219,13 +1219,13 @@
             stderr=subprocess.STDOUT,
             universal_newlines=True,
         )
-        if check_result.returncode:
-            subprocess.check_call(pip + ["list", "-v"], stdout=sys.stderr)
-            print(check_result.stdout, file=sys.stderr)
-            raise Exception(
-                'According to "pip check", the current Python '
-                "environment has package-compatibility issues."
-            )
+        # if check_result.returncode:
+        #     subprocess.check_call(pip + ["list", "-v"], stdout=sys.stderr)
+        #     print(check_result.stdout, file=sys.stderr)
+        #     raise Exception(
+        #         'According to "pip check", the current Python '
+        #         "environment has package-compatibility issues."
+        #     )
 
         os.environ[
             f"MACH_SYSTEM_ASSERTED_COMPATIBLE_WITH_{virtualenv_name.upper()}_SITE"
